hooks/GNS.js

```diff
@@ -0,0 +1,193 @@
+        0         1       2 3        4 5    6  7   8   9   10  11  12
+        |         |       | |        | |    |  |   |   |   |   |   |
+ $--GNS,hhmmss.ss,ddmm.mm,a,dddmm.mm,a,c--c,xx,x.x,x.x,x.x,x.x,x.x*hh<CR><LF>
+------------------------------------------------------------------------------
+Field Number:
+0. UTC of position, hh is hours, mm is minutes, ss.ss is seconds
+1. Latitude, dd is minutes, mm.mm is minutes
+2. N or S (North or South)
+3. Longitude, dd is minutes, mm.mm is minutes
+4. E or W (East or West)
+5. Mode indicator (non-null) - Variable character field with one character for each supported constellation:
+    * First character is for GPS
+    * Second character is for GLONASS
+    * Third character is Galileo
+    * Fourth character is for BeiDou
+    * Fifth character is for QZSS
+    * Subsequent characters will be added for new constellations
+   Each character will be one of the following
+     - N = No fix. Satellite system not used in position fix, or fix not valid
+     - A = Autonomous. Satellite system used in non-differential mode in position fix
+     - D = Differential (including all OmniSTAR services). Satellite system used in differential mode in position fix
+     - P = Precise. Satellite system used in precision mode. Precision mode is defined as: no deliberate degradation (such as Selective Availability) and higher resolution code (P-code) is used to compute position fix
+     - R = Real-Time Kinematic. Satellite system used in RTK mode with fixed integers
+     - F = Float RTK. Satellite system used in real-time kinematic mode with floating integers
+     - E = Estimated (dead reckoning) mode
+     - M = Manual Input mode
+     - S = Simulator mode
+6. Total number of satellites in use, 00-99
+7. Horizontal Dilution of Precision (HDOP), calculated using all the satellites (GPS, GLONASS, and any future satellites) and used in computing the solution reported in each GNS sentence
+8. Antenna altitude, meters, re:mean-sea-level (geoid)
+9. Goeidal separation meters - The difference between the earth ellipsoid surface and mean-sea-level (geoid) surface defined by the reference datum used in the position solution
+10. Age of differential data - Null if talker ID is GN, additional GNS messages follow with Age of differential data
+11. Differential reference station ID, 0000-4095 - Null if Talker ID is GN, Additional GNS messages follow with Reference station ID
+12. Navigational status (added when the IEC61162-1:2010/NMEA 0183 V4.10 option is selected in the NMEA I/O configuration):
+     - S = Safe
+     - C = Caution
+     - U = Unsafe
+     - V = Not valid for navigation
+13. Checksum
+*/
+
+function isEmpty(mixed) {
+  return (
+    (typeof mixed !== 'string' && typeof mixed !== 'number') ||
+    (typeof mixed === 'string' && mixed.trim() === '')
+  )
+}
+
+function indicator(chars, modes) {
+    const systems = ["GPS","GNSS","Galileo","BeiDou","QZSS"]
```

Comment:
GNSS => Global Navigation Satellite System, this should be GLONASS here.

But I think decoding the pretty compact and machine readable string to a single human oriented string is a disservice. We should have something like 
```
{ 
   GPS: 'Precise',
   Galileo: 'Autonomous'
}
```
- only include the ones that are operational - we don't want to list all of them in each GNS message, if the poor GNSS device has only GPS for example
- define short but human readable enumerated strings for the modes
- use an object structure to capture this data instead of a string
-